src/pages/shop/shop.tsx

```diff
@@ -1,12 +1,30 @@
 import { axiosClient } from "../../utils/axios";
+import { useQuery } from "@tanstack/react-query";
 
-async function testAPI() {
-  return await axiosClient.get("/test");
+async function listSupplements() {
+  return await axiosClient.get("/supplements");
 }
 
+type supplement = {
+  name: string;
+  slug: string;
+  category: string;
+  description: string;
+  photo_url: string;
+  unit_price: number;
+  benefits: string[];
+};
+
 function Shop() {
-  testAPI().then(console.log).catch(console.error);
-  return <div>Shop</div>;
+  const query = useQuery(["supplements"], async () => await listSupplements());
```

Comment:
could this provide `listSupplements` as the callback function directly since it's returning a Promise? that might look like `const query = useQuery(["supplements"], listSupplements);`